File: Teme-TPA/house_robber.py
class Solution(object):
    def rob(self, nums):
        k=0
        if(len(nums)==1):
            return nums[0]
        if(len(nums)==2):
            if(nums[0]>=nums[1]):
                return nums[0]
            else:
                return nums[1]
        nums[1]=max(nums[0],nums[1])
        for i in range(2,len(nums)):
            nums[i]=max(nums[i-2]+nums[i],nums[i-1])
        return nums[len(nums)-1]

File: Teme-TPA/house_robber.c
int max(int a,int b)
{
    if(a>=b)
        return a;
    else
        return b;
}
int rob(int* nums, int numsSize) 
{   
    if(numsSize==1)
        return nums[0];
    if(numsSize==2)
    {
        if(nums[0]<nums[1])
            return nums[1];
        else
            return nums[0]; 
    }
    nums[1]=max(nums[0],nums[1]);
    for(int i=2;i<numsSize;i++)
    {
        nums[i]=max(nums[i-2]+nums[i],nums[i-1]);
    }
    return nums[numsSize-1];
}